### PR TITLE
RDM-8382: Allow configuring an event specific default value for a com…

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
RDM-8382: Allow configuring an event specific default value for a complex type subfield
    
            [RDM-8382] (https://tools.hmcts.net/jira/browse/RDM-8382).
            Allow configuring an event specific default value for a complex type subfield

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java [hunk 1]
```diff
@@ -43,9 +53,54 @@
...
+                            return false;
+                        }
+                    }
+                ).collect(Collectors.toList()))
+        );
+    }
+
+    private Optional<String> getOrganisationPolicyDefaultRoleValue(final CaseDataContent content) {
+
+        final JsonNode existingDat  = new ObjectMapper().convertValue(content.getData(), JsonNode.class);
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java:86]: better use the standard object mapper in the related utils class

File: src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java [hunk 2]
```diff
@@ -43,9 +53,54 @@
...
+                            return validateOrgPolicyCaseAssignedRole(caseEventFieldComplexDefinition, defaultValue.get());
+                        } else {
+                            return false;
+                        }
+                    }
+                ).collect(Collectors.toList()))
+        );
+    }
+
+    private Optional<String> getOrganisationPolicyDefaultRoleValue(final CaseDataContent content) {
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java:84]: there may be many default values!

File: src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java [hunk 3]
```diff
@@ -43,9 +53,54 @@
...
+
+        final JsonNode existingDat  = new ObjectMapper().convertValue(content.getData(), JsonNode.class);
+        final List<JsonNode> jsonNode = existingDat.findParents(ORGANISATION_POLICY_ROLE);
+        final  Optional<JsonNode> node = jsonNode.stream().findFirst();
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java:88]: what is multiple node are there?

**Comment by MSancaktutar** [comment]
Please make sure we have a green build after all.

File: src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java (hunk 3 above)
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java]: DONE

File: src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java (hunk 2 above)
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java]: DOBE

**Pull request merged**